File: EngActivator/EngActivator.APP/Dtos/ActivityResponse/ActivityResponseForCreate.cs
using EngActivator.APP.Shared.Enums;

namespace EngActivator.APP.Dtos
{
    public class ActivityResponseForCreate
    {
        public string Answer { get; set; }
        public string Activity { get; set; }
        public ActivityTypeEnum ActivityTypeId { get; set; }
    }
}

File: eng_activator_app/lib/models/activity_response/activity_response_for_create.dart
import 'dart:convert';
import 'package:eng_activator_app/models/activity/activity.dart';
import 'package:eng_activator_app/shared/enums.dart';

class ActivityResponseForCreate {
  String answer;
  Activity activity;
  ActivityTypeEnum activityTypeId;

  ActivityResponseForCreate({
    required String answer,
    required Activity activity,
    required ActivityTypeEnum activityTypeId,
  })  : answer = answer,
        activity = activity,
        activityTypeId = activityTypeId;

  Map<String, dynamic> toJson() {
    return {
      "answer": answer,
      "activity": jsonEncode(activity),
      "activityTypeId": activityTypeId.index + 1,
    };
  }
}
